src/libraries/Microsoft.PowerFx.Core/Public/DeclarationsCheckResult.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            Contracts.AssertValue(symbols);
+
+            if (_symbols != null)
+            {
+                throw new InvalidOperationException($"Can only call {nameof(SetBindingInfo)} once.");
+            }
+
+            _symbols = symbols;
+
+            return this;
+        }
+
+        internal DeclarationsCheckResult SetText(string definitions, ParserOptions parserOptions = null)
+        {
+            Contracts.AssertValue(definitions);
+
+            if (_definitions != null)
+            {
+                throw new InvalidOperationException($"Can only call {nameof(SetText)} once.");
+            }
+
+            _definitions = definitions;
+            _parserOptions = parserOptions ?? new ParserOptions();
+            _defaultErrorCulture = _parserOptions.Culture ?? CultureInfo.InvariantCulture;
+
+            return this;
+        }
+
+        internal ParseUserDefinitionResult ApplyParse()
+        {
+            if (_definitions == null)
+            {
+                throw new InvalidOperationException($"Must call {nameof(SetText)} before calling ApplyParse().");
+            }
+
+            if (_parse == null)
+            {
+                _parse = UserDefinitions.Parse(_definitions, _parserOptions);
+
+                if (_parse.HasErrors)
+                {
+                    _errors.AddRange(ExpressionError.New(_parse.Errors, _defaultErrorCulture));
+                }
+            }
+
+            return _parse;
+        }
+
+        public IReadOnlyDictionary<DName, FormulaType> ResolvedTypes => ResolveTypes();
+
+        internal IReadOnlyDictionary<DName, FormulaType> ResolveTypes()
+        {
+            if (_parse == null || _parse.DefinedTypes == null || _symbols == null)
+            {
+                throw new InvalidOperationException($"Must call {nameof(ApplyParse)} with valid defined types and {nameof(SetBindingInfo)} before calling ResolveTypes().");
+            }
+
+            if (_resolvedTypes == null)
+            {
+                if (_parse.DefinedTypes.Any())
+                {
+                    this._resolvedTypes = DefinedTypeResolver.ResolveTypes(_parse.DefinedTypes, _symbols, out var errors);
+                    _errors.AddRange(ExpressionError.New(errors, _defaultErrorCulture));
+                }
+                else
+                {
+                    this._resolvedTypes = ImmutableDictionary<DName, FormulaType>.Empty;
+                }
+            }
+
+            return this._resolvedTypes;
+        }
+
+        // All errors accumulated. 
+        private readonly List<ExpressionError> _errors = new List<ExpressionError>();
+
+        /// <summary>
+        /// List of all errors and warnings. Check <see cref="ExpressionError.IsWarning"/>.
+        /// This can include Parse, ResolveType errors />,
+        /// Not null, but empty on success.
+        /// </summary>
+        public IEnumerable<ExpressionError> Errors => GetErrorsInLocale(null);
```

Comment:
Add ApplyErrors()  - this does everything that could produce an error! 

ok to have _error collection filled out incrementally.